pynxtools/dataconverter/convert.py

```diff
@@ -118,6 +154,56 @@ def convert(input_file: Tuple[str, ...],
         **kwargs,
     )
     helpers.validate_data_dict(template, data, nxdl_root)
+    return data
+
+
+# pylint: disable=too-many-arguments,too-many-locals
+def convert(input_file: Tuple[str, ...],
+            reader: str,
+            nxdl: str,
+            output: str,
+            generate_template: bool = False,
+            fair: bool = False,
+            undocumented: bool = False,
+            **kwargs):
+    """The conversion routine that takes the input parameters and calls the necessary functions.
+
+    Parameters
+    ----------
+    input_file : Tuple[str]
+        Tuple of files or file
+    reader: str
+        Name of reader such as xps
+    nxdl : str
+        Root name of nxdl file, e.g. NXmpes for NXmpes.nxdl.xml
+    output : str
+        Output file name.
+    generate_template : bool, default False
+        True if user wants template in logger info.
+    fair : bool, default False
+        If True, a warning is given that there are undocumented paths
+        in the template.
+    undocumented : bool, default False
+        If True, an undocumented warning is given.
+
+    Returns
+    -------
+    None.
+    """
+
+    nxdl_root, nxdl_path = get_nxdl_root_and_path(nxdl)
```

Comment:
Please change name nxdl_path to nxdl_file_path or something. We use nxdl_path in various places in the converter code and it refers to a /path/in/nxdl/structure rather than an filesystem path.

Also, this nxdl_path is not needed. It's only being used in the writer to get the nxdl_root again. So there is no point in passing the path. We can remove this.

Reply:
Addresed, new_name -> new_f_name.

Reply:
I think we can even remove it completely. If you follow the change, it's just going to the writer to get the NXroot again... We can just pass the NXroot there and it should still work.

Reply:
addressed!

Reply:
It's not addressed. The function still sends back the nxdl_f_path and the Writer still takes the path to get the XML root element again. 

But I will let it slide. I'll fix it myself later.